server/apps/archive/archive_ingest.py

```diff
@@ -63,10 +64,15 @@ def create(self, docs, **kwargs):
             else:
                 # We are fetching from ingest
                 ingest_doc = superdesk.get_resource_service('ingest').find_one(req=None, _id=doc.get('guid'))
+
                 if not ingest_doc:
                     msg = 'Fail to found ingest item with guid: %s' % doc.get('guid')
                     raise SuperdeskApiError.notFoundError(msg)
 
+                if doc.get('macro'):
+                    # there is a macro so transform it
+                    ingest_doc = get_resource_service('macros').execute_macro(ingest_doc, doc.get('macro'))
+
```

Comment:
No, there wasn't. I have moved the transformation after the check of state transition.